eng/pipelines/templates/variables/globals.yml

```diff
@@ -1,6 +1,6 @@
 variables:
   OfficialBuildId: $(Build.BuildNumber)
   skipComponentGovernanceDetection: true
-  DotNetCoreVersion: '3.1.405'
+  DotNetCoreVersion: '6.0.403'
```

Comment:
We are also going to need to update our target frameworks in the various projects.

Reply:
We have also talked about moving this version into https://github.com/Azure/azure-sdk-tools/blob/main/global.json similar to what azure-sdk-for-net repo does.

Reply:
@weshaggard 

> We are also going to need to update our target frameworks in the various projects.

I found more issues related to this caused by SDKs going out of support. Please see #4934 and #4935.

However, once these problems are solved, I don't think that `TargetFramework` has to exactly match the SDK used, as long as the targeted framework is not newer than the SDK and is not obsolete. Which admittedly is a bit moot point now, because everything before `net6.0` is obsolete (-:

> We have also talked about moving this version into https://github.com/Azure/azure-sdk-tools/blob/main/global.json similar to what azure-sdk-for-net repo does.

How about I do the migration to `global.json` in a separate, follow-up PR?

Also, I think this PR is no longer required to unblock PR [#4915 per my comment there](https://github.com/Azure/azure-sdk-tools/pull/4915#issuecomment-1345059377), so I suggest we go ahead with merging #4915 to mitigate #4888, and then work on more broad fix proposed by this PR.